compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/service/validator/ServiceValidator.java

```diff
@@ -132,6 +138,75 @@ public void validate() {
         } else if (serviceNode.kind() == SyntaxKind.OBJECT_CONSTRUCTOR) {
             validateServiceObject();
         }
+        validateEntities();
+    }
+
+    private void validateEntities() {
+        for (Map.Entry<String, Symbol> entry : this.interfaceEntityFinder.getEntities().entrySet()) {
+            AnnotationSymbol entityAnnotationSymbol = getEntityAnnotationSymbol(entry.getValue());
+            if (entityAnnotationSymbol == null) {
+                continue;
+            }
+            AnnotationNode annotationNode = getEntityAnnotationNode(entityAnnotationSymbol, entry.getKey());
+            if (annotationNode == null) {
+                continue;
+            }
+            validateEntityAnnotation(annotationNode);
+        }
+    }
+
+    private AnnotationNode getEntityAnnotationNode(AnnotationSymbol annotationSymbol, String entityName) {
+        Project currentProject = this.context.currentPackage().project();
+        EntityAnnotationFinder entityAnnotationFinder = new EntityAnnotationFinder(this.context.semanticModel(),
+                                                                                   currentProject,
+                                                                                   this.context.moduleId(),
+                                                                                   annotationSymbol, entityName);
+        return entityAnnotationFinder.find().orElse(null);
+    }
+
+    private void validateEntityAnnotation(AnnotationNode entityAnnotation) {
+        if (entityAnnotation.annotValue().isEmpty()) {
+            return;
+        }
+        for (MappingFieldNode fieldNode : entityAnnotation.annotValue().get().fields()) {
+            if (fieldNode.kind() != SPECIFIC_FIELD) {
+                addDiagnostic(CompilationDiagnostic.PROVIDE_KEY_VALUE_PAIR_FOR_ENTITY_ANNOTATION, fieldNode.location());
+                continue;
+            }
+            SpecificFieldNode specificFieldNode = (SpecificFieldNode) fieldNode;
+            Node fieldNameNode = specificFieldNode.fieldName();
+            if (fieldNameNode.kind() != SyntaxKind.IDENTIFIER_TOKEN) {
+                continue;
+            }
+            IdentifierToken fieldNameToken = (IdentifierToken) fieldNameNode;
+            String fieldName = fieldNameToken.text().trim();
```

Comment:
Do we need the `trim()` here?

Reply:
The trim method removes the leading or trailing minutiae.